### PR TITLE
V1 Todo project

### PR DESCRIPTION
**Pull request by osamaakb**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (osamaakb) [inline index.html:57]: remove this white space

File: main.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+let noTasks = document.getElementById('noTasks');
+
+// Getting today's date
+  n =  new Date();
+  y = n.getFullYear();
+  m = n.getMonth() + 1;
+  d = n.getDate();
+  document.getElementById("currentDate").innerHTML = m + "/" + d + "/" + y; 
+
+  window.onload = inputBox.focus();
```
Comment (osamaakb) [inline main.js:11]: Nice addetion

File: main.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  d = n.getDate();
+  document.getElementById("currentDate").innerHTML = m + "/" + d + "/" + y; 
+
+  window.onload = inputBox.focus();  
+
+  addTodoBtn.onclick = function addTask() { 
+  let inputValue = document.getElementById("inputValue").value; 
+  let deadlineDate = document.getElementById("deadlineDate").value; 
+  if (inputValue === '') 
+  {
```
Comment (osamaakb) [inline main.js]:
```suggestion
  if (inputValue === '') {
```

File: main.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    alert("This field cannot be empty!");
+  }
+  else
+  {
```
Comment (osamaakb) [inline main.js]: pay attention to the indentation here, the `{` should be in front of the `else` like this
```suggestion
  else {
```

File: main.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    let li = document.createElement('li');
+    li.className = 'li';
+    let textElement = document.createElement('span');
+    let dateElement = document.createElement('span');
+    textElement.textContent = inputValue;
+    dateElement.textContent = deadlineDate;
+    dateElement.className = 'date-element';
+    li.appendChild(textElement);
+    li.appendChild(dateElement);
+    ul.appendChild(li);
```
Comment (osamaakb) [inline main.js:31]: this way of creating new html element and assigning data to it work fine, I personally prefer the wiring HTML string and inserting it directly more like this:
``` html
<li>
   <span>{textContent}</span>
   <span class="date-element">{dateContent}</span>
</li>
```

then insert this HTML to your document using `innerHTML` of `insertAdjacentHTML`, both works fine

File: main.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+    // Clears the inputValue on click
```
Comment (osamaakb) [inline main.js:33]: the comment here isn't really needed, it is kind of clear what this line of code is doing (:

File: style.css [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+}
+
+.li {
+padding: 8px;
+}
+
+#noTasks {
+  text-align: center;
+}
+#currentDate {
```
Comment (osamaakb) [inline style.css:16]: add an empty line between each css block

File: style.css [hunk 8]
```diff
@@ -0,0 +1,23 @@
+.selector-for-some-widget {
+  box-sizing: content-box;
+}
+span.date-element {
+padding: 25px;
+}
```
Comment (osamaakb) [inline style.css]:
```suggestion
  box-sizing: content-box;
}

span.date-element {
  padding: 25px;
}
```

**Review by osamaakb — COMMENTED** [review]
There are a lot of bootstrap css and js files that are not required, you can remove them.
Same goes for the j-query file isn't needed.

to use bootstrap you only need to import it with a single line without it's files
refer to this link for more info 
https://getbootstrap.com/docs/5.0/getting-started/introduction/

it is true that it may take a little time to load but for the sake of organization and not importing a ton of files let's do it as the link above (:

File: main.js (hunk 5 above)
Comment (MayceAlazzawi) [reply main.js:31]: I searched for how to do it using the way you suggested but I didn't find a good solution, can tell me what should I search for to learn how to use your way?

Comment (osamaakb) [reply main.js:31]: Here are some readings about that, I searched for "how to use insertadjacenthtml with lists in JS"

https://www.javascripttutorial.net/javascript-dom/javascript-insertadjacenthtml/
https://developer.mozilla.org/en-US/docs/Web/API/Element/insertAdjacentHTML

**Pull request merged**